src/Request.ts

```diff
@@ -413,10 +413,8 @@ export class Request extends Macroable implements RequestContract {
    * The value of trustProxy is passed directly to [proxy-addr](https://www.npmjs.com/package/proxy-addr)
    */
   public protocol (): string {
-    const protocol = this.parsedUrl.protocol!
-
     if (!this._config.trustProxy(this.request.connection.remoteAddress!, 0)) {
-      return protocol
+      return 'http'
```

Comment:
I think always returning `http` is bit weird, so we can check for `this.request.connection.encrypted` and return `https`. https://nodejs.org/dist/latest-v12.x/docs/api/tls.html#tls_tlssocket_encrypted. 

For some reason, the types of Node.js doesn't have this flag, but it does exists at runtime when using the `https` server of Node. So this is what I suggest.

1. Write a test using the `https` server
2. Ensure that protocol returns `https`, since the `encrypted` flag will be available.

Also the `encrypted` flag should get priority over everything else

Reply:
Funny, I initially implemented it with `this.request.connection.encrypted` but I wasn't able to make  a test case so I reverted the change.
If you have an idea to tell supertest to use TLS, I'll redo it.

Reply:
`supertest` takes an instance of the http server, that means, we can pass in an instance of `https` server instance. I suggest using [pem](https://www.npmjs.com/package/pem) to generate the `secret key` and `certificate`.

#### Untested code

```ts
import { promisify } from 'util'
import { createServer: httpsServer } from 'https'
import { createCertificate } from 'pem'

const keys = promisify(createCertificate({ days: 1, selfSigned: true }))

const server = httpsServer({
 key: keys.serviceKey,
 cert: keys.certificate,
}, function (req, res) {
  
})

supertest(server)
```

Reply:
I added a commit to show you the problem.
Even if I use `createServer` from `https`, supertest doesn't know that it should use TLS to connect to it. The only way I can think of would be to use a full URL for the request such as `https://localhost/` but I don't know which port number to use (supertest binds to a random port).

Reply:
This is what I have used to run the test

```js
    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
    createCertificate({}, async (error, keys) => {
      if (error) {
        done(error)
        return
      }

      const server = httpsServer({ key: keys.serviceKey, cert: keys.certificate }, (req, res) => {
        const request = new Request(req, res, fakeConfig())
        res.writeHead(200, { 'content-type': 'application/json' })
        res.end(JSON.stringify({ secure: request.secure() }))
      })

      const { body } = await supertest(server).get('/')
      assert.deepEqual(body, {
        secure: false,
      })
      done()
    })
```

We have to look for `req.connection['encrypted']` instead

Reply:
done